minwoo.lee_java/src/BOJ2138.java

```diff
@@ -0,0 +1,65 @@
+import java.io.*;
+import java.util.Arrays;
+
+public class BOJ2138 {
+    static int result = Integer.MAX_VALUE;
+    static int N;
+
+    public static void main(String[] args) throws IOException {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        N = Integer.parseInt(br.readLine());
+        boolean[] start = new boolean[N];
+        boolean[] end = new boolean[N];
+        makeBulb(start, br.readLine());
+        makeBulb(end, br.readLine());
+        boolean[] startCopy = Arrays.copyOf(start, start.length);
+        // 첫번째 스위치 눌렀을 때
+        start[0] = !start[0];
+        start[1] = !start[1];
+        solution(start, end, 1, 1);
+        // 첫번째 스위치를 누르지 않았을 때
+        solution(startCopy, end, 0, 1);
+
+        StringBuilder sb = new StringBuilder();
+        System.out.println((result == Integer.MAX_VALUE) ? sb.append(-1) : sb.append(result));
```

Comment:
```suggestion
        System.out.println((result == Integer.MAX_VALUE) ? -1 : result);
```
`StringBuilder`를 하나만 `append`하고 바로 사용하고 끝나는데, 위와 같이 쓰는게 좀 더 좋지 않았을까 싶습니다!